src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/Contracts/S3CredentialRequestParams.cs

```diff
@@ -0,0 +1,35 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#nullable disable
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+using Microsoft.SqlTools.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectManagement.Contracts
+{
+    public class NewS3CredentialRequestParams : GeneralRequestDetails
+    {
+        /// <summary>
+        /// URI of the underlying connection for this request
+        /// </summary>
+        public string S3Url { get; set; }
+
+        /// <summary>
+        /// Secret to access S3 storage
+        /// </summary>
+        public string Secret { get; set; }
+        /// <summary>
+        /// Target database
+        /// </summary>
+        public string ConnectionUri { get; set; }
+    }
+
+    public class NewS3CredentialRequestResponse { }
+
+    public class NewS3CredentialRequest
+    {
+        public static readonly RequestType<NewS3CredentialRequestParams, NewS3CredentialRequestResponse> Type = RequestType<NewS3CredentialRequestParams, NewS3CredentialRequestResponse>.Create("objectManagement/newS3CredentialRequest");
```

Comment:
Instead of making this S3 specific what about just updating the base object management credential stuff as needed and using that? From what I can tell there's nothing here that's "special" - it's just a normal credential with a specific name (which can easily be specified on the UI side instead)

Reply:
And then for the GetS3Credentials - I'd suggest making a generic message that can get a credential with a specific name and using that instead (if such a thing doesn't already exist). That way it's usable for other things too if needed in the future.